src/java.desktop/share/classes/javax/swing/BorderFactory.java

```diff
@@ -153,7 +155,10 @@ public static Border createBevelBorder(int type) {
      * @return the <code>Border</code> object
      */
     public static Border createBevelBorder(int type, Color highlight, Color shadow) {
-        return new BevelBorder(type, highlight, shadow);
+        if (highlight != null && shadow != null) {
+            return new BevelBorder(type, highlight, shadow);
+        }
+        return new BevelBorder(type);
```

Comment:
Do we need to do something similar for other border types such as EtchedBorder ?